homeassistant/components/ecoforest/entity.py

```diff
@@ -0,0 +1,42 @@
+"""Base Entity for Ecoforest."""
+from __future__ import annotations
+
+from pyecoforest.models.device import Device
+
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity import EntityDescription
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN, MANUFACTURER
+from .coordinator import EcoforestCoordinator
+
+
+class EcoforestEntity(CoordinatorEntity[EcoforestCoordinator]):
+    """Common Ecoforest entity using CoordinatorEntity."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        coordinator: EcoforestCoordinator,
+        description: EntityDescription,
+    ) -> None:
+        """Initialize device information."""
+        self.entity_description = description
+        self._attr_unique_id = f"{DOMAIN}_{description.key}"
```

Comment:
Oh forgot to mention, please make them really unique:
```suggestion
        self._attr_unique_id = f"{coordinator.data.serial_number}_{description.key}"
```

Reply:
you could add DOMAIN in it, but I don't see any real reason why you want that